### DISCUSSION TITLE
Change control mode in Python for Torque mode

### DISCUSSION THREAD

**Discussion by Timozegu** [robotology/community#632] [Q&A] [answered]
Dear Robotology community

I need to move the torque value of the different joints using python.
It was working using C++ but i can't seem to make it work on Python , as VOCAB_CM_TORQUE variable (Vocab32) cannot seem to be called.
Do you have an idea on how i can make the robots, joint's torque move?

Thank you,

Kind regards,

Tim

**Comment by traversaro** [comment]
> It was working using C++ but i can't seem to make it work on Python , as VOCAB_CM_TORQUE variable (Vocab32) cannot seem to be called. Do you have an idea on how i can make the robots, joint's torque move?

Can you provide the code that you are tryng to run and the exact error you are getting? Thanks!

**Comment by Timozegu** [comment]
hello ! 
here is the code i am trying to work with  ( the goal being to move the 4th joint of the right arm) : 

```python
import yarp;
import time;
import numpy as np;

yarp.Network.init();
 
props = yarp.Property()
props.put("device","remote_controlboard")
props.put("local","/client/right_arm")
props.put("remote","/icubSim/right_arm")
  
armDriver = yarp.PolyDriver(props)
 
iPos = armDriver.viewIPositionControl()
iVel = armDriver.viewIVelocityControl()
iTorque = armDriver.viewITorqueControl()
iEnc = armDriver.viewIEncoders()
iimp = armDriver.viewIImpedanceControl()
ictrl = armDriver.viewIControlMode()

jnts=iPos.getAxes()

iimp.setImpedance(4, 0.111, 0.014) 

encs=yarp.Vector(jnts)
iEnc.getEncoders(encs.data())
  
ictrl.setControlMode(4,VOCAB_CM_TORQUE);
jnt_torque = 3
iTorque.setRefTorque(4,jnt_torque)
```
this is based on the python tutorial.
and i get as an error : 
```python
Traceback (most recent call last):
  File "/home/timozegu/code/python/python-motor-control.py", line 35, in <module>
    ictrl.setControlMode(4,VOCAB_CM_TORQUE);
NameError: name 'VOCAB_CM_TORQUE' is not defined
```
it is working fine on c++ where do almost the same thing : 
```c++
 ictrl->setControlMode(4,VOCAB_CM_TORQUE);
```

Thank you,

Tim

**Comment by traversaro** [comment, marked as answer]
In C++ `VOCAB_CM_TORQUE` is a preprocessor definition, so it is used without namespace. However, in Python it is wrapped as a normal constant, so it needs to be used as `yarp.VOCAB_CM_TORQUE`, see https://github.com/roboticslab-uc3m/tools/blob/878c99f6ee48051665afe487b17d6b5a415a18c7/programs/pidTuning.py#L80 or https://github.com/robotology/blender-robotics-utils/blob/8839079844acb6c3a91860276b083a9a06a4be4c/script/blenderRCBPanel/blenderRCBPanel.py#L121 for an example of the use of `setControlMode` in Python.

**Comment by Timozegu** [comment]
thank you very much for your answer, it worked ! 

Thank you :)

Tim
